File: RoghtiaSystem_JahidKawa.Server/Data/Models/Patient.cs
namespace RoghtiaSystem_JahidKawa.Server.Data.Models
{
    public class Patient
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public int Age { get; set; }
        public string Gender { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public int UserId { get; set; }
        public Users User { get; set; } = null!;
    }
}

File: roghtiasystem_jahidkawa.client/shared/types/patient.ts
export interface PatientInput {
  name: string
  phone: string
  age: number
  gender: string
  address: string
}

export interface Patient extends PatientInput {
  code: string
  id: number
  user: { id: number, userName: string }
}

export interface PatientPage {
  items: Patient[]
  total: number
  page: number
  pageSize: number
}

export interface PatientQuery {
  page?: number
  pageSize?: number
  search?: string
}
